tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -695,6 +695,81 @@ class Restrictions {
     return errors;
   }
 
+  List<SourceSpanSeverityException> validateFieldType(
+    TypeDefinition fieldType,
+    SourceSpan? span,
+  ) {
+    var typeText = span?.text;
+    if (typeText != null && _isNoValidationType(typeText)) return [];
+
+    var errors = <SourceSpanSeverityException>[];
+
+    if (_isUnsupportedType(fieldType)) {
+      errors.add(SourceSpanSeverityException(
+        'The datatype "${fieldType.className}" is not supported in models.',
+        span,
+      ));
+      return errors;
+    }
+
+    var moduleAlias = fieldType.moduleAlias;
+    if (_isUnresolvedModuleType(fieldType) && moduleAlias != null) {
```

Comment:
Suggestion (NIT): Switch the order of evaluation here.

Module alias seems more likely to abort our evaluation immediately rather than checking if it is an unresolved module type.